modules/price/src/Plugin/Field/FieldFormatter/PriceCalculatedFormatter.php

```diff
@@ -84,11 +84,7 @@ public static function create(ContainerInterface $container, array $configuratio
    * {@inheritdoc}
    */
   public function viewElements(FieldItemListInterface $items, $langcode) {
-    $currency_codes = [];
-    foreach ($items as $delta => $item) {
-      $currency_codes[] = $item->currency_code;
-    }
-    $currencies = $this->currencyStorage->loadMultiple($currency_codes);
+    $currencies = $this->currencyStorage->loadMultiple();
```

Comment:
Yeah, I agree most cases it'll be 1 field with 1 currency. But we need to support any use cases :)

I see what you're saying, we shouldn't load the currency from the field value, but rather from

```
      $resolved_price = $this->chainPriceResolver->resolve($purchasable_entity, '1');
      $number = $resolved_price->getNumber();
+   $currency = $this->currencyStorage->load($number->getCurrencyCode());
```